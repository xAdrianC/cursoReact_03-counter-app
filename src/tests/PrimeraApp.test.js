import { render } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';


describe('Pruebas en primeraApp', () => {
    // test('Debe devolver "hola, soy Adri"', () =>{
    //     const saludo = 'Hola, soy Adri';
    //     const {getByText} = render(<PrimeraApp saludo={saludo}/>);
    //     expect( getByText(saludo)).toBeInTheDocument();
    // });

    test('debe mostrar <PrimeraApp /> correctamente ', () => {
        const saludo = 'Hola, soy Adri';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar el subtitulo enviado pro props ', () => {
        const saludo = 'Hola, soy Adri';
        const subtitulo = 'Hola, soy el subtitulo de adri';
        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);
    })
    
    
});