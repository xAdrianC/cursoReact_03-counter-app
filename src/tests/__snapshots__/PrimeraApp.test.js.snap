// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primeraApp debe mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola, soy Adri
  </h1>
  <p>
    Mi primera app
  </p>
</Fragment>
`;
